lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1107,9 +1107,9 @@ public function resetQueryPart($queryPartName)
      */
     private function getSQLForSelect()
     {
-        $query = 'SELECT ' . implode(', ', $this->sqlParts['select']) . ' FROM ';
+        $query = 'SELECT ' . implode(', ', $this->sqlParts['select']);
 
-        $query .= implode(', ', $this->getFromClauses())
+        $query .= ($this->sqlParts['from'] ? ' FROM ' . implode(', ', $this->getFromClauses()) : '')
```

Comment:
Yeah it is neither meant to be platform aware nor is it meant to be able to validate SQL..